tokendefault.php

```diff
@@ -219,63 +219,57 @@ function tokendefault_civicrm_mosaicoConfig(&$config) {
 
 function tokendefault_civicrm_mosaicoScriptUrlsAlter(&$scriptUrls) {
   $res = CRM_Core_Resources::singleton();
-
-  $coreResourceList = $res->coreResourceList('html-header');
-  $coreResourceList = array_filter($coreResourceList, 'is_string');
-  foreach ($coreResourceList as $item) {
+  $snippets = Civi::service('bundle.coreResources')->getAll();
+  foreach ($snippets as $snippet) {
+    $itemUrl = NULL;
     if (
-      FALSE !== strpos($item, 'js')
-      && !strpos($item, 'crm.menubar.js')
-      && !strpos($item, 'crm.wysiwyg.js')
-      && !strpos($item, 'l10n-js')
-    ) {
-      if ($res->isFullyFormedUrl($item)) {
-        $itemUrl = $item;
+        FALSE !== strpos($snippet['name'], 'js')
+        && !strpos($snippet['name'], 'crm.menubar.js')
+        && !strpos($snippet['name'], 'crm.menubar.min.js')
+        && !strpos($snippet['name'], 'crm.wysiwyg.js')
+        && !strpos($snippet['name'], 'crm.wysiwyg.min.js')
+        && !strpos($snippet['name'], 'l10n-js')
+      ) {
+      if ($snippet['scriptUrl'] ?? FALSE) {
+        $itemUrl = $snippet['scriptUrl'];
+      }
+      elseif ($snippet['scriptFile'] && count($snippet['scriptFile']) == 2) {
+        $itemUrl = $res->getUrl($snippet['scriptFile'][0], $snippet['scriptFile'][1], TRUE);
       }
-      else {
-        $item = CRM_Core_Resources::filterMinify('civicrm', $item);
-        $itemUrl = $res->getUrl('civicrm', $item, TRUE);
+      if ($itemUrl) {
+        $scriptUrls[] = $itemUrl;
       }
-      $scriptUrls[] = $itemUrl;
     }
   }
 
-  // Include our own JS.
-  $url = $res->addCacheCode(CRM_Utils_System::url('civicrm/tokendefault/mosaico-js', '', TRUE, NULL, NULL, NULL, NULL));
+  // Include our own JS (this url generates dynammic JS containing apprpopriate settings per session).
+  $url = CRM_Utils_System::url('civicrm/tokendefault/mosaico-js', '', TRUE, NULL, NULL, NULL, NULL);
   $scriptUrls[] = $url;
 }
 
 function tokendefault_civicrm_mosaicoStyleUrlsAlter(&$styleUrls) {
   $res = CRM_Core_Resources::singleton();
+  $snippets = array_merge(Civi::service('bundle.coreResources')->getAll(), Civi::service('bundle.coreStyles')->getAll());
 
-  // Load custom or core css
-  $config = CRM_Core_Config::singleton();
-  if (!Civi::settings()->get('disable_core_css')) {
-    $styleUrls[] = $res->getUrl('civicrm', 'css/civicrm.css', TRUE);
-  }
-  if (!empty($config->customCSSURL)) {
-    $customCSSURL = $res->addCacheCode($config->customCSSURL);
-    $styleUrls[] = $customCSSURL;
-  }
   // crm-i.css added ahead of other styles so it can be overridden by FA.
   array_unshift($styleUrls, $res->getUrl('civicrm', 'css/crm-i.css', TRUE));
```

Comment:
$styleUrls is undefined. Either this line is unnecessary and should be removed, or it's incorrect and should be fixed. @lucky091588 can you address this?

Reply:
@twomice See latest commit for the fix. I added back the code in line [255](https://github.com/twomice/com.joineryhq.tokendefault/pull/4/commits/61beedd9cb7b9cbd63b7f24eff5fd29240632609#diff-d60323bbcc3cfb8458807a627bea3777907ef4025e9ddf134e6f70c41ba75e1bR255) to 262